src/endpoint/scheduler.rs

```diff
@@ -370,6 +374,36 @@ impl JobHandle {
     }
 }
 
+impl Drop for JobHandle {
+    fn drop(&mut self) {
+        debug!("Cleaning up JobHandle");
+        if self.container_id.is_some() {
+            debug!("Container was already started");
+            let docker = self.endpoint.docker().clone();
+            let container_id = self.container_id.take().unwrap();
+
+            tokio::spawn(async move {
+                let container = docker.containers().get(&container_id);
+                let container_info = container.inspect().await.unwrap();
+
+                if container_info.state.running {
+                    debug!("Container is still running, cleaning up...");
+                    match container.kill(None).await {
+                        Ok(_) => debug!("Stopped container with id: {}", container_id),
+                        Err(e) => {
+                            error!("Failed to stop container with id: {}\n{}", container_id, e)
+                        }
+                    }
+                } else {
+                    debug!("Container has already finished");
+                }
+            });
+        } else {
+            debug!("No container created");
```

Comment:
Sounds weird since we aren't creating containers here. I'd be more explicit, e.g.: "No container was created for the JobHandle yet -> skipping cleanup".

Reply:
Sounds good. Is the approach getting the state of the container in drop implementation ok?
Or should we set the container id back to None again like you wrote in the other comment?

Reply:
I would set the `container_id` back to `None` again as that is much cleaner (in this case already for the debug output but not setting it back to `None` could create bigger issues in the future if we start using it in other places - therefore you should always keep it up-to-date).